core/embed/trezorhal/stm32f4/xdisplay/st-7789/display_fb.c

```diff
@@ -63,45 +63,58 @@ void display_physical_fb_clear(void) {
 }
 
 #ifndef BOARDLOADER
-static volatile bool pending_fb_switch = false;
+static volatile uint16_t pending_fb_switch = 0;
 static volatile uint32_t last_fb_update_time = 0;
 #endif
 
 #ifndef BOARDLOADER
 void DISPLAY_TE_INTERRUPT_HANDLER(void) {
-  HAL_NVIC_DisableIRQ(DISPLAY_TE_INTERRUPT_NUM);
-
-  if (current_frame_buffer == 1) {
-    bg_copy_start_const_out_8((uint8_t *)physical_frame_buffer_1,
-                              (uint8_t *)DISPLAY_DATA_ADDRESS,
-                              DISPLAY_RESX * DISPLAY_RESY * 2);
-
-  } else {
-    bg_copy_start_const_out_8((uint8_t *)physical_frame_buffer_0,
-                              (uint8_t *)DISPLAY_DATA_ADDRESS,
-                              DISPLAY_RESX * DISPLAY_RESY * 2);
+  if (pending_fb_switch == 1) {
```

Comment:
It seems to me that this new algorithm (waiting for the TE interrupt twice) can slow down `display_refresh()`, which might prevent us from achieving animation at full framerate. Am I right?

Reply:
Probably yes. But i needed to prevent situation when two fast refreshes happened and then the first frame was not fully drawn on the display. Theoretically, i think one TE should be enough to show the frame - but observations told different story. Perhaps there is another problem and we should investigate this further when we have time.